azurerm/internal/services/network/tests/express_route_circuit_peering_resource_test.go

```diff
@@ -311,6 +368,188 @@ resource "azurerm_express_route_circuit_peering" "test" {
+  peering_location      = "Silicon Valley"
+  bandwidth_in_mbps     = 50
+
+  sku {
+    tier   = "Premium"
+    family = "MeteredData"
+  }
+
+  tags = {
+    Env     = "Test"
+    Purpose = "AcceptanceTests"
+  }
+}
+
+resource "azurerm_express_route_circuit_peering" "test" {
+  peering_type                  = "MicrosoftPeering"
+  express_route_circuit_name    = azurerm_express_route_circuit.test.name
+  resource_group_name           = azurerm_resource_group.test.name
+  peer_asn                      = 100
+  primary_peer_address_prefix   = "192.168.3.0/30"
+  secondary_peer_address_prefix = "192.168.4.0/30"
+  vlan_id                       = 300
+
+  microsoft_peering_config {
+    advertised_public_prefixes = ["123.2.0.0/24"]
+  }
+
+  ipv6_peering {
+    primary_peer_address_prefix   = "2002:db01::/126"
+    secondary_peer_address_prefix = "2003:db01::/126"
+
+    microsoft_peering_config {
+      advertised_public_prefixes = ["2002:db01::/126"]
+    }
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func testAccAzureRMExpressRouteCircuitPeering_msPeeringIpv6CustomerRouting(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_express_route_circuit" "test" {
+  name                  = "acctest-erc-%d"
+  location              = azurerm_resource_group.test.location
+  resource_group_name   = azurerm_resource_group.test.name
+  service_provider_name = "Equinix"
+  peering_location      = "Silicon Valley"
+  bandwidth_in_mbps     = 50
+
+  sku {
+    tier   = "Premium"
+    family = "MeteredData"
+  }
+
+  tags = {
+    Env     = "Test"
+    Purpose = "AcceptanceTests"
+  }
+}
+
+resource "azurerm_express_route_circuit_peering" "test" {
+  peering_type                  = "MicrosoftPeering"
+  express_route_circuit_name    = azurerm_express_route_circuit.test.name
+  resource_group_name           = azurerm_resource_group.test.name
+  peer_asn                      = 100
+  primary_peer_address_prefix   = "192.168.3.0/30"
+  secondary_peer_address_prefix = "192.168.4.0/30"
+  vlan_id                       = 300
+
+  microsoft_peering_config {
+    advertised_public_prefixes = ["123.2.0.0/24"]
+  }
+  ipv6_peering {
+    primary_peer_address_prefix   = "2002:db01::/126"
+    secondary_peer_address_prefix = "2003:db01::/126"
+
+    microsoft_peering_config {
+      advertised_public_prefixes = ["2002:db01::/126"]
+      customer_asn               = 64511
+      routing_registry_name      = "ARIN"
+    }
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func testAccAzureRMExpressRouteCircuitPeering_msPeeringIpv6WithRouteFilter(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```

Comment:
could we use this for the resource group names:
```suggestion
  name     = "acctestRG-expressroute-%d"
```